platform/common/core/generic/vault/rwset.go

```diff
@@ -90,7 +91,7 @@ func (r NamespaceWrites) Equals(o NamespaceWrites) error {
 
 type Writes map[string]NamespaceWrites
 
-func (r Writes) equals(o Writes, nss ...string) error {
+func (r Writes) Equals(o Writes, nss ...string) error {
 	return entriesEqual(r, o, func(a, b NamespaceWrites) bool { return a.Equals(b) == nil }, nss...)
```

Comment:
In the end we need a set of nss to make the search more efficient. We can calculate it already and pass the set instead of a slice.

Reply:
do you want to apply the changes directly?

Reply:
okay